src/services/mongodb/collection.js

```diff
@@ -112,140 +108,97 @@ class Collection {
   aggregate(pipeline) {
     return aggregateOp(this, pipeline);
   }
+}
 
-  /**
-   * Gets the number of documents matching the filter.
-   *
-   * @param {Object} query The query used to match documents.
-   * @param {Object} options Additional find options.
-   * @param {Number} [options.limit=null] The maximum number of documents to return.
-   * @return {Number} An array of documents.
-   */
-  count(query, options = {}) {
-    return findOp(this, query, Object.assign({}, options, {
-      count: true
-    }), result => !!result.result ? result.result[0] : 0);
-  }
+// private
+function setObjectId(doc) {
+  if (doc._id === undefined || doc._id === null) doc._id = new ObjectID();
+  return doc;
+}
 
-  // deprecated
-  insert(docs, options = {}) {
-    return insertOp(this, docs, options);
-  }
+function insertOneOp(service, doc) {
+  const outgoingOptions = { document: setObjectId(doc) };
 
-  upsert(query, update, options = {}) {
-    return updateOp(this, query, update, Object.assign({}, options, { upsert: true }));
-  }
+  return buildResponse('insertOne', service, buildArgs(service, outgoingOptions));
 }
 
-// deprecated methods
-Collection.prototype.upsert =
-  deprecate(Collection.prototype.upsert, 'use `updateOne`/`updateMany` instead of `upsert`');
+function insertManyOp(service, docs) {
+  docs = Array.isArray(docs) ? docs : [ docs ];
 
-// private
-function insertOp(self, docs, options) {
-  let stages = [];
-
-  // there may be no docs, when building for chained pipeline stages in
-  // which case the source is considered to be the previous stage
-  if (docs) {
-    docs = Array.isArray(docs) ? docs : [ docs ];
-
-    // add ObjectIds to docs that have none
-    docs = docs.map(doc => {
-      if (doc._id === undefined || doc._id === null) doc._id = new ObjectID();
-      return doc;
-    });
-
-    stages.push({
-      action: 'literal',
-      args: {
-        items: docs
-      }
-    });
+  const outgoingOptions = { documents: docs.map(setObjectId) };
```

Comment:
we don't actually need to do this anymore - the server now assigns object id's to the docs being inserted - can remove.